src/parser.ts

```diff
@@ -18,16 +19,18 @@ export class Parser {
   public readonly parserRegistry = new Map<string, SchemaParser>();
   public readonly spectral: Spectral;
 
-  constructor(options?: ParserOptions) {
-    const { spectral } = options || {};
+  constructor(
+    private readonly options?: ParserOptions
+  ) {
+    const { spectral } = this.options || {};
     if (spectral instanceof Spectral) {
       this.spectral = spectral;
     } else {
       this.spectral = new Spectral(spectral);
     }
 
-    // TODO: fix type
-    this.spectral.setRuleset(aasRuleset as any);
+    this.registerSchemaParser(AsyncAPISchemaParser());
```

Comment:
You can now register the other ones as all have been merged.

Reply:
No. As current solution, people need to register given format manually. For example, RAML works only on nodejs, so register RAML here will break browser compability.

Reply:
Oh, that's right! 👍  thanks